### PR TITLE
Clippy: integration Vol. 2

### PR DESCRIPTION
**Pull request by s373r**
## Description

Related issue: https://github.com/kamu-data/kamu-cli/issues/332

## Checklist before requesting a review
- [x] [CHANGELOG.md](./CHANGELOG.md) updated
- [x] API changes are backwards-compatible
- [x] Workspace layout changes include a migration
- [x] Documentation update PR: <link or N/A>
- [x] Dataset pipelines update scheduled if needed

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -96,22 +105,20 @@ codegen-engine-tonic:
...
 codegen-graphql:
 	python $(ODF_SPEC_DIR)/tools/jsonschema_to_rust_gql.py $(ODF_SPEC_DIR)/schemas \
-		| $(RUSTFMT) > src/adapter/graphql/src/scalars/odf_generated.rs
+		> src/adapter/graphql/src/scalars/odf_generated.rs
 	$(call add_license_header, "src/adapter/graphql/src/scalars/odf_generated.rs")
 
 
 .PHONY: codegen
 codegen: codegen-odf-serde codegen-engine-tonic codegen-graphql
+	cargo fmt --all
```
Comment (sergiimk) [inline Makefile:121]: what was the reason for not piping through `rustfmt`?

this change will mean that if I run `make codegen-graphql` I will have a poorly formatted file

(although it's very rare that I run individual targets and not the top-level `codegen`)

File: src/app/cli/src/cli_commands.rs [hunk 2]
```diff
@@ -90,10 +90,7 @@ pub fn get_command(
             cli_catalog.get_one()?,
             validate_many_dataset_refs(
                 cli_catalog,
-                submatches
-                    .get_many("dataset")
-                    .unwrap() // required
-                    .map(|r: &DatasetRef| r.clone()),
+                submatches.get_many("dataset").unwrap().cloned(),
```
Comment (sergiimk) [inline src/app/cli/src/cli_commands.rs:93]: Oh watch out for the clap stuff and make sure to test every command affected. I think this may cause runtime panic.

File: Makefile (hunk 1 above)
Comment (s373r) [reply Makefile:121]: Honesty, just to cover all files since these lines didn't perform actual formatting
```shell
	$(RUSTFMT) $(ODF_CRATE_DIR)/src/engine/grpc_generated/engine.rs
	$(RUSTFMT) $(ODF_CRATE_DIR)/src/engine/grpc_generated/engine.tonic.rs
```

> this change will mean that if I run make codegen-graphql I will have a poorly formatted file

makes sense, I'll fix that

File: src/app/cli/src/cli_commands.rs (hunk 2 above)
Comment (s373r) [reply src/app/cli/src/cli_commands.rs:93]: hm, interesting -- yes, will recheck just in case

File: src/infra/core/src/utils/docker_images.rs [hunk 3]
```diff
@@ -7,18 +7,17 @@
 // the Business Source License, use of this software will be governed
 // by the Apache License, Version 2.0.
 
-pub const SPARK: &'static str = "ghcr.io/kamu-data/engine-spark:0.22.1-spark_3.1.2";
-pub const FLINK: &'static str =
-    "ghcr.io/kamu-data/engine-flink:0.18.0-flink_1.16.0-scala_2.12-java8";
-pub const DATAFUSION: &'static str = "ghcr.io/kamu-data/engine-datafusion:0.7.0";
+pub const SPARK: &str = "ghcr.io/kamu-data/engine-spark:0.22.1-spark_3.1.2";
```
[the remark this answers is not in the document]
Comment (s373r) [reply src/infra/core/src/utils/docker_images.rs:10]: yeah, was paying attention to that too when I was revising -- but it's no longer necessary now

File: src/infra/ingest-datafusion/benches/snapshot.rs [hunk 4]
```diff
@@ -69,29 +69,29 @@ async fn setup(
     let mut rng = rand::rngs::SmallRng::seed_from_u64(123127986998);
 
     rng.try_fill(&mut buf[..]).unwrap();
-    pk1.append_slice(&buf[orig_range.clone()]);
+    pk1.append_slice(&buf[orig_range]);
     new_pk1.append_slice(&buf[new_range.clone()]);
```
[the remark this answers is not in the document]
Comment (s373r) [reply src/infra/ingest-datafusion/benches/snapshot.rs:73]: `RangeTo` has `Copy` trait 
```rust
#[lang = "RangeTo"]
#[doc(alias = "..")]
#[derive(Copy, Clone, PartialEq, Eq, Hash)]
#[stable(feature = "rust1", since = "1.0.0")]
pub struct RangeTo<Idx> {
    /// The upper bound of the range (exclusive).
    #[stable(feature = "rust1", since = "1.0.0")]
    pub end: Idx,
}
```

`RangeFrom`, by some reason, at the moment nope
```rust
#[lang = "RangeFrom"]
#[doc(alias = "..")]
#[derive(Clone, PartialEq, Eq, Hash)] // not Copy -- see #27186
#[stable(feature = "rust1", since = "1.0.0")]
pub struct RangeFrom<Idx> {
    /// The lower bound of the range (inclusive).
    #[stable(feature = "rust1", since = "1.0.0")]
    pub start: Idx,
}
```

File: src/domain/opendatafabric/src/dtos/dtos_generated.rs [hunk 5]
```diff
@@ -443,38 +438,30 @@ impl_enum_variant!(MergeStrategy::Ledger(MergeStrategyLedger));
 
 /// Snapshot merge strategy.
 ///
-/// This strategy can be used for data dumps that are taken periodical
-/// and contain only the latest state of the observed entity or system.
-/// Over time such dumps can have new rows added, and old rows either removed
-/// or modified.
+/// This strategy can be used for data state snapshots that are taken
```
[the remark this answers is not in the document]
Comment (s373r) [reply src/domain/opendatafabric/src/dtos/dtos_generated.rs]: Rebase, done

File: Makefile (hunk 1 above)
Comment (s373r) [reply Makefile]: returned piping, don

File: src/app/cli/src/cli_commands.rs (hunk 2 above)
Comment (s373r) [reply src/app/cli/src/cli_commands.rs:93]: checked: worked as expected

File: src/infra/ingest-datafusion/benches/snapshot.rs (hunk 4 above)
[the remark this answers is not in the document]
Comment (sergiimk) [reply src/infra/ingest-datafusion/benches/snapshot.rs:73]: I think Rust 2024 edition is planning to do fix `Range` types to make them all `Copy`

File: Makefile [hunk 6]
```diff
@@ -69,16 +79,16 @@ codegen-odf-serde:
 	python $(ODF_SPEC_DIR)/tools/jsonschema_to_flatbuffers.py $(ODF_SPEC_DIR)/schemas > $(ODF_CRATE_DIR)/schemas/odf.fbs
 	flatc -o $(ODF_CRATE_DIR)/src/serde/flatbuffers --rust --gen-onefile $(ODF_CRATE_DIR)/schemas/odf.fbs
 	mv $(ODF_CRATE_DIR)/src/serde/flatbuffers/odf_generated.rs $(ODF_CRATE_DIR)/src/serde/flatbuffers/proxies_generated.rs
-	$(RUSTFMT) $(ODF_CRATE_DIR)/src/serde/flatbuffers/proxies_generated.rs
+	rustfmt $(ODF_CRATE_DIR)/src/serde/flatbuffers/proxies_generated.rs
```
Comment (sergiimk) [inline Makefile:82]: I think we should revert these changes.

Previous code was passing `--edition 2021` option to `rustfmt`. The default edition is still `2015` [according to the docs](https://rust-lang.github.io/rustfmt/?version=v1.7.0&search=#edition), and I distinctly remember having some formatting issues because of the edition mismatch.

**Review by sergiimk — COMMENTED** [review]
Looks good to me except for one comment.

Will defer merge decision to @zaychenko-sergei.

File: Makefile (hunk 6 above)
Comment (s373r) [reply Makefile:82]: We have `rustfmt.toml` with specified `edition = "2021"`

Also, I double-checked that the file is read if `rustfmt` is run outside of `cargo fmt`

**Review by zaychenko-sergei — APPROVED** [review]
